src/m365/file/commands/file-move.ts

```diff
@@ -0,0 +1,240 @@
+
+interface CommandArgs {
+  options: Options;
+}
+
+interface Options extends GlobalOptions {
+  webUrl: string;
+  sourceUrl: string;
+  targetUrl: string;
+  newName?: string;
+  nameConflictBehavior?: string;
+}
+
+class FileMoveCommand extends GraphCommand {
+  private pollingInterval: number = 10_000;
+  private readonly nameConflictBehaviorOptions = ['fail', 'replace', 'rename'];
+
+  public get name(): string {
+    return commands.MOVE;
+  }
+
+  public get description(): string {
+    return 'Moves a file to another location using the Microsoft Graph';
+  }
+
+  constructor() {
+    super();
+
+    this.#initTelemetry();
+    this.#initOptions();
+    this.#initValidators();
+  }
+
+  #initTelemetry(): void {
+    this.telemetry.push((args: CommandArgs) => {
+      Object.assign(this.telemetryProperties, {
+        webUrl: typeof args.options.webUrl !== 'undefined',
+        sourceUrl: typeof args.options.sourceUrl !== 'undefined',
+        targetUrl: typeof args.options.targetUrl !== 'undefined',
+        newName: typeof args.options.newName !== 'undefined',
+        nameConflictBehavior: typeof args.options.nameConflictBehavior !== 'undefined'
+      });
+    });
+  }
+
+  #initOptions(): void {
+    this.options.unshift(
+      { option: '-u, --webUrl <webUrl>' },
+      { option: '-s, --sourceUrl <sourceUrl>' },
+      { option: '-t, --targetUrl <targetUrl>' },
+      { option: '--newName [newName]' },
+      { option: '--nameConflictBehavior [nameConflictBehavior]', autocomplete: this.nameConflictBehaviorOptions }
+    );
+  }
+
+  #initValidators(): void {
+    this.validators.push(
+      async (args: CommandArgs) => {
+        if (args.options.nameConflictBehavior && this.nameConflictBehaviorOptions.indexOf(args.options.nameConflictBehavior) === -1) {
+          return `${args.options.nameConflictBehavior} is not a valid nameConflictBehavior value. Allowed values: ${this.nameConflictBehaviorOptions.join(', ')}.`;
+        }
+
+        return validation.isValidSharePointUrl(args.options.webUrl);
+      }
+    );
+  }
+
+  public async commandAction(logger: Logger, args: CommandArgs): Promise<void> {
+    try {
+      const { webUrl, sourceUrl, targetUrl, nameConflictBehavior, newName, verbose } = args.options;
+      const sourcePath: string = this.getAbsoluteUrl(webUrl, sourceUrl);
+      const destinationPath: string = this.getAbsoluteUrl(webUrl, targetUrl);
+
+      if (verbose) {
+        logger.logToStderr(`Moving file '${sourcePath}' to '${destinationPath}'...`);
+      }
+
+      const { siteId, driveId, itemId } = await this.getDriveIdAndItemId(webUrl, sourcePath, sourceUrl, logger, verbose);
+      const { targetDriveId, targetItemId } = await this.getTargetDriveAndItemId(webUrl, targetUrl, logger, verbose);
+      const requestOptions: CliRequestOptions = this.getRequestOptions(targetDriveId, targetItemId, newName, sourcePath);
+
+      const isSameDrive: boolean = driveId === targetDriveId;
+      const apiUrl =
```

Comment:
Let's just build the URL string within the getRequestOptions function. That's where all the request related magic occurs.